routers/api/v1/repo/key.go

```diff
@@ -75,26 +75,29 @@ func ListDeployKeys(ctx *context.APIContext) {
 	//   "200":
 	//     "$ref": "#/responses/DeployKeyList"
 
-	var keys []*models.DeployKey
-	var err error
+	opts := &models.ListDeployKeysOptions{
+		ListOptions: utils.GetListOptions(ctx),
+		RepoID:      ctx.Repo.Repository.ID,
+		KeyID:       ctx.QueryInt64("key_id"),
+		Fingerprint: ctx.Query("fingerprint"),
+	}
 
-	fingerprint := ctx.Query("fingerprint")
-	keyID := ctx.QueryInt64("key_id")
-	if fingerprint != "" || keyID != 0 {
-		keys, err = models.SearchDeployKeys(ctx.Repo.Repository.ID, keyID, fingerprint)
-	} else {
-		keys, err = models.ListDeployKeys(ctx.Repo.Repository.ID, utils.GetListOptions(ctx))
+	keys, err := models.ListDeployKeys(opts)
```

Comment:
So there is a Xorm function called FindAndCount which doesn't pass the limits which may be better.

Reply:
oh didn't know if it ... will need to refactor a lot then

Reply:
**FindAndCount** exec two db querys so we should only use it if the function is always used next to a count function